### PR TITLE
Interactivity API: Fix interactivity api e2e tests.

### PR DESCRIPTION
**Pull request by cbravobernal**
## What?

Update the priority of removing the SSR for e2e tests to the number according to the current Core one.

### CODE REVIEW

File: lib/compat/wordpress-6.5/interactivity-api/interactivity-api.php [hunk 1]
```diff
@@ -58,16 +58,20 @@ function wp_interactivity_process_directives_of_interactive_blocks( array $parse
...
 
 				/*
-				* Uses a priority of 20 to ensure that other filters can add additional
-				* directives before the processing starts.
-				*/
-				add_filter( 'render_block_' . $block_name, $process_interactive_blocks, 20, 2 );
+				 * Uses a priority of 100 to ensure that other filters can add additional
+				 * directives before the processing starts.
+				 */
+				add_filter( 'render_block_' . $block_name, $process_interactive_blocks, 100, 2 );
```
Comment (sirreal) [inline lib/compat/wordpress-6.5/interactivity-api/interactivity-api.php:64]: [Matches core](https://github.com/WordPress/wordpress-develop/blob/0c24793d257cc4ce2f929a6e911f6b94bf97f19a/src/wp-includes/interactivity-api/interactivity-api.php#L63) 👍

File: lib/compat/wordpress-6.5/interactivity-api/interactivity-api.php [hunk 2]
```diff
@@ -58,16 +58,20 @@ function wp_interactivity_process_directives_of_interactive_blocks( array $parse
...
 		}
 
 		return $parsed_block;
 	}
-	add_filter( 'render_block_data', 'wp_interactivity_process_directives_of_interactive_blocks' );
+	/*
+	 * Uses a priority of 100 to ensure that other filters can edit $parsed_block
+	 * without crashing the SSR.
+	 */
+	add_filter( 'render_block_data', 'wp_interactivity_process_directives_of_interactive_blocks', 100 );
```
Comment (sirreal) [inline lib/compat/wordpress-6.5/interactivity-api/interactivity-api.php]: [core has 1 as the last argument here](https://github.com/WordPress/wordpress-develop/blob/0c24793d257cc4ce2f929a6e911f6b94bf97f19a/src/wp-includes/interactivity-api/interactivity-api.php#L69-L73), although I think that's the default and maybe it should be omitted in core.

**Review by sirreal — APPROVED** [review]

File: lib/compat/wordpress-6.5/interactivity-api/interactivity-api.php (hunk 2 above)
Comment (cbravobernal) [reply lib/compat/wordpress-6.5/interactivity-api/interactivity-api.php]: True

Comment (swissspidy) [reply lib/compat/wordpress-6.5/interactivity-api/interactivity-api.php]: Yes it should be omitted, pointless to add the default

**Comment by getdave** [comment]
I think this PR probably should have been backported to `wp/6.5` branch to avoid tests breaking there also. Going to look into doing that.

**Comment by getdave** [comment]
I just manually cherry picked this PR into the `wp/6.5` branch in https://github.com/WordPress/gutenberg/commit/b9e73ef9be4a742e82dd16373646074e1047f01f in order to get the tests passing in case we need further release for WP 6.5.

**Pull request merged**